### DISCUSSION TITLE
Multiple upload instances, if/else

### DISCUSSION THREAD

**Discussion by mnoesel** [uploadcare/file-uploader#337] [blocks] [answered]
Hi,

I'd need support to make this script work for multiple uploader instances on one site.
This script works to get the cdnUrls from a single uploader on a site. But I have four different upload instances on a site.

Their CTX-Names are this:

Uploaders: uploader-1, uploader-2, uploader-3 and uploader-4
Variables: upload1, upload2, upload3, upload4

```
  let outputElement = document.querySelector('lr-data-output');
  
  outputElement.addEventListener('lr-data-output', (e) => {
    console.log(e.detail);
    // or
    console.log(outputElement.value);
    //code to get the cdnUrls and push them to an array variable named upload1
    var upload1 = [];
    e.detail.data.forEach(function(v){
    upload1.push(v['cdnUrl']);
    });
  });
```

This should be very easy for someone who can code.
Hope somebody can help.

Greets.

**Comment by nd0ut** [comment, marked as answer]
Hey @mnoesel,

you need to use separate `lr-data-output` blocks per uploader.


Or you could use `LR_UPLOAD_FINISH` global event (described [here](https://uploadcare.github.io/blocks/solutions/file-uploader/#events)) like this:

```js
window.addEventListener('LR_UPLOAD_FINISH', (e) => {
  const ctxName = e.detail.ctx
  const files = e.detail.data
  // check for ctxName and push files where you need
});
```
